src/sentry/testutils/cases.py

```diff
@@ -1223,15 +1225,16 @@ def store_metric(
     ):
         internal_metric = METRICS_MAP[metric]
         entity = self.ENTITY_MAP[metric]
-        if tags is None:
-            tags = {}
-        else:
-            tags = {
-                indexer.resolve(self.organization.id, key): indexer.resolve(
-                    self.organization.id, value
-                )
-                for key, value in tags.items()
-            }
+        org_id = self.organization.id
+        transformed_tags: Dict[int, int] = {}
+
+        if tags:
+            for k, v in tags.items():
+                resolved_key = indexer.resolve(org_id, k)
+                resolved_value = indexer.resolve(org_id, v)
+                new_key = resolved_key if resolved_key else indexer.record(org_id, k)
+                new_value = resolved_value if resolved_value else indexer.record(org_id, v)
+                transformed_tags[new_key] = new_value
```

Comment:
We should not be sending `None` tags to snuba - we can create the record if it doesn't already exist

Reply:
Should be sufficient to replace `indexer.resolve` with `indexer.record` though, right? Since `record` basically does "resolve or record".

Reply:
yes good point, will update